recipes/icu/all/conanfile.py

```diff
@@ -134,19 +138,23 @@ def _workaround_icu_20545(self):
     @property
     def _build_config_cmd(self):
         prefix = self.package_folder.replace("\\", "/")
-        platform = {("Windows", "Visual Studio"): "Cygwin/MSVC",
-                    ("Windows", "gcc"): "MinGW",
-                    ("AIX", "gcc"): "AIX/GCC",
-                    ("AIX", "xlc"): "AIX",
-                    ("SunOS", "gcc"): "Solaris/GCC",
-                    ("Linux", "gcc"): "Linux/gcc",
-                    ("Linux", "clang"): "Linux",
-                    ("Macos", "gcc"): "MacOSX",
-                    ("Macos", "clang"): "MacOSX",
-                    ("Macos", "apple-clang"): "MacOSX",
-                    ("FreeBSD", "gcc"): "FreeBSD",
-                    ("FreeBSD", "clang"): "FreeBSD"}.get((str(self.settings.os),
-                                                          str(self.settings.compiler)))
+        platform = {
+            ("Windows", "Visual Studio"): "Cygwin/MSVC",
+            ("Windows", "gcc"): "MinGW",
+            ("AIX", "gcc"): "AIX/GCC",
+            ("AIX", "xlc"): "AIX",
+            ("SunOS", "gcc"): "Solaris/GCC",
+            ("Linux", "gcc"): "Linux/gcc",
+            ("Linux", "clang"): "Linux",
+            ("Macos", "gcc"): "MacOSX",
+            ("Macos", "clang"): "MacOSX",
+            ("Macos", "apple-clang"): "MacOSX",
+            ("iOS", "apple-clang"): "MacOSX",
+            ("tvOS", "apple-clang"): "MacOSX",
+            ("watchOS", "apple-clang"): "MacOSX",
+            ("FreeBSD", "gcc"): "FreeBSD",
+            ("FreeBSD", "clang"): "FreeBSD",
+        }.get((str(self.settings.os), str(self.settings.compiler)))
```

Comment:
we need to improve an error, might do in this PR or another PR, as you wish
something like:
1. extracting `platform` to a property
2. in validate add some code like:
```
def validate(self):
    if self.settings.os is not in self._platform.keys():
        raise ConanInvalidConfiguration("unsupported platform, give up")
```

Reply:
I agree

Reply:
done